### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by kb6673** [mhzawadi/foxess-mqtt#23] [General]
Thank you for this. I really struggled to get data out of Fox Inverter, I tried all sorts and started going down the road of creating logins programmatically and I them I discovered this docker image - worked first time, instantly out of the box.

---

I'd like to suggest / help with a few things. I'd like the option to supply my own MQTT topic. 
Do you think we could pop a new config var into `config.json` ?

**Comment by mhzawadi** [comment]
Hey @kb6673 

Glad you found this, will see what I can do about the topic name.

**Comment by mhzawadi** [comment]
Hello 👋🏼 

I have been working on your request and my dev image `mhzawadi/foxess-mqtt:dev-php` has working code in it, please try that and see if that is what you need.

mqtt_topic: this is a custom top level topic, you will still get sub topics for each device on your account

**Comment by kb6673** [comment]
`$this->mqtt->post_mqtt(''.$mqtt_topic.'/'.$deviceSN.'/'.$name, $value);` is pretty much exactly how I was doing it too :) You beat me to a PR :)

**Reply by mhzawadi** [reply]
nice, if your good with PHP then please have a read and PR any improvements

**Comment by mhzawadi** [comment]
v0.0.7 is out, please test it and if you have feedback or bugs please open an issue
